### PR TITLE
v1.0.0 beta release : Native Auth Support

### PR DESCRIPTION
**Pull request by newbreedofgeek**
v1.0.0 beta release. backwards compatible native auth for view data, deep param validations for mint and access data, comment out custom error methods and repalce with pure Error as they were hiding error details on client side catch, updated documentation.

- https://github.com/Itheum/sdk-mx-data-nft/issues/30
- https://github.com/Itheum/sdk-mx-data-nft/issues/25
- https://github.com/Itheum/sdk-mx-data-nft/issues/22 (updated and tested logic)
- Also tested this PR in real integration use case : https://github.com/Itheum/sdk-mx-data-nft/pull/29

### CODE REVIEW

File: src/utils.ts [hunk 1]
```diff
@@ -91,10 +91,444 @@ export async function checkTraitsUrl(traitsUrl: string) {
...
+  stream?: boolean | undefined;
+  fwdAllHeaders?: boolean | undefined;
+  fwdHeaderKeys?: string | undefined;
+  mvxNativeAuthEnable?: number | undefined;
+  mvxNativeAuthAccessToken?: string | undefined;
+  mvxNativeAuthMaxExpirySeconds?: number | undefined;
+  mvxNativeAuthOrigins?: string[] | undefined;
+  fwdHeaderMapLookup?: any;
+  _fwdHeaderMapLookupMustContainBearerAuthHeader?: boolean | undefined;
+  _mandatoryParamsList: string[]; // a pure JS fallback way to validate mandatory params, as typescript rules for mandatory can be bypassed by client app
```
Comment (bucurdavid) [inline src/utils.ts:106]: I would exclude the `_mandatoryParamsList` from params and add it as a separate param to use the advantage of object assigning directly

i.e: 
```typescript
let params  = {signedMessage?: string | undefined,
  signableMessage?: any,
  stream?: boolean | undefined,
  fwdAllHeaders?: boolean | undefined,
  fwdHeaderKeys?: string | undefined,
  mvxNativeAuthEnable?: number | undefined,
  mvxNativeAuthAccessToken?: string | undefined,
  mvxNativeAuthMaxExpirySeconds?: number | undefined,
  mvxNativeAuthOrigins?: string[] | undefined,
  fwdHeaderMapLookup?: any,
  _fwdHeaderMapLookupMustContainBearerAuthHeader?: boolean | undefined,
  };
validateSpecificParamsViewData(params, ['signedMessage', ... ]);
```

File: src/datanft.ts [hunk 2]
```diff
@@ -235,54 +246,73 @@ export class DataNft {
...
+
+    // S: run any format specific validation
+    const { allPassed, validationMessages } = validateSpecificParamsViewData({
+      signedMessage: p.signedMessage,
+      signableMessage: p.signableMessage,
+      stream: p.stream,
+      fwdAllHeaders: p.fwdAllHeaders,
+      fwdHeaderKeys: p.fwdHeaderKeys,
+      fwdHeaderMapLookup: p.fwdHeaderMapLookup,
+      _mandatoryParamsList: ['signedMessage', 'signableMessage']
```
[the remark this answers is not in the document]
Comment (newbreedofgeek) [reply src/datanft.ts:297]: good suggestion

[the remark this answers is not in the document]
Comment (newbreedofgeek) [reply src/datanft.ts:297]: but let's make this as an update on the next version so we can test it fully locally... as i've heavily tested the current version locally.

File: src/datanft.ts [hunk 3]
```diff
@@ -312,26 +342,146 @@ export class DataNft {
...
+        fwdHeaderKeys: p.fwdHeaderKeys,
+        fwdHeaderMapLookup: p.fwdHeaderMapLookup,
+        fwdAllHeaders: p.fwdAllHeaders,
+        stream: p.stream,
+        _fwdHeaderMapLookupMustContainBearerAuthHeader: true,
+        _mandatoryParamsList: [
+          'mvxNativeAuthOrigins',
+          'mvxNativeAuthMaxExpirySeconds',
+          'fwdHeaderMapLookup'
+        ]
```
[the remark this answers is not in the document]
Comment (newbreedofgeek) [reply src/datanft.ts:420]: but let's make this as an update on the next version so we can test it fully locally... as i've heavily tested the current version locally.

[the remark this answers is not in the document]
Comment (newbreedofgeek) [reply src/datanft.ts:420]: _fwdHeaderMapLookupMustContainBearerAuthHeader could also move to a new param to keep it cleaner

File: src/utils.ts (hunk 1 above)
Comment (newbreedofgeek) [reply src/utils.ts:106]: as above

**Pull request merged**